packages/opentelemetry-plugin-mongodb/src/mongodb.ts

```diff
@@ -0,0 +1,255 @@
+    shimmer.massUnwrap([this._moduleExports.Server.prototype], this
+      .SERVER_FNS as never[]);
+    shimmer.massUnwrap([this._moduleExports.Cursor.prototype], this
+      .CURSOR_FNS_FIRST as never[]);
+  }
+
+  /** Creates spans for Command operations */
+  private getPatchCommand() {
+    const plugin = this;
+    return (original: Func<mongodb.Server>) => {
+      return function(
+        this: mongodb.Server,
+        ns: string,
+        command: MongoInternalCommand,
+        options: {} | Function,
+        callback: Func<unknown>
+      ): mongodb.Server {
+        const currentSpan = plugin._tracer.getCurrentSpan();
+        if (currentSpan === null) {
+          return original.apply(this, (arguments as unknown) as unknown[]);
+        }
+        const resultHandler =
+          typeof options === 'function' ? options : callback;
+        if (typeof resultHandler !== 'function') {
+          return original.apply(this, (arguments as unknown) as unknown[]);
+        }
+        if (typeof command !== 'object') {
+          return original.apply(this, (arguments as unknown) as unknown[]);
+        }
+        let type: string;
+        if (command.createIndexes) {
+          type = 'createIndexes';
+        } else if (command.findandmodify) {
+          type = 'findAndModify';
+        } else if (command.ismaster) {
+          type = 'isMaster';
+        } else if (command.count) {
+          type = 'count';
+        } else {
+          type = 'command';
+        }
+
+        const span = plugin._tracer.startSpan(`${ns}.${type}`, {
+          parent: currentSpan,
+          kind: SpanKind.CLIENT,
+        });
+        if (typeof options === 'function') {
+          return original.call(
+            this,
+            ns,
+            command,
+            plugin.patchEnd(span, options as Func<unknown>)
+          );
+        } else {
+          return original.call(
+            this,
+            ns,
+            command,
+            options,
+            plugin.patchEnd(span, callback)
+          );
+        }
+      };
+    };
+  }
+
+  /** Creates spans for Query operations */
+  private getPatchQuery() {
+    const plugin = this;
+    return (original: Func<mongodb.Server>) => {
+      return function(
+        this: mongodb.Server,
+        ns: string,
+        command: MongoInternalCommand,
+        options: {},
+        callback: Func<unknown>
+      ): mongodb.Server {
+        const currentSpan = plugin._tracer.getCurrentSpan();
+        if (currentSpan === null) {
+          return original.apply(this, (arguments as unknown) as unknown[]);
+        }
+        const resultHandler =
+          typeof options === 'function' ? options : callback;
+        if (typeof resultHandler !== 'function') {
+          return original.apply(this, (arguments as unknown) as unknown[]);
+        }
+        const span = plugin._tracer.startSpan(`${ns}.query`, {
+          kind: SpanKind.CLIENT,
```

Comment:
I am not sure if this should be `SpanKind.SERVER` or `SpanKind.CLIENT`.

Reply:
As per the specs, `SpanKind` MUST be `Client`. https://github.com/open-telemetry/opentelemetry-specification/blob/c2846dea5b3a8d636d6b65130890a2ce998451a9/specification/data-semantic-conventions.md#databases-client-calls